src/jit/gentree.cpp

```diff
@@ -1740,6 +1740,35 @@ bool GenTreeCall::IsHelperCall(Compiler* compiler, unsigned helper) const
     return IsHelperCall(compiler->eeFindHelper(helper));
 }
 
+void GenTreeCall::ReplaceCallOperand(GenTree** operandUseEdge, GenTree* replacement)
+{
+    assert(operandUseEdge != nullptr);
+    assert(replacement != nullptr);
+    assert(TryGetUse(*operandUseEdge, &operandUseEdge));
+
+    GenTree* originalOperand = *operandUseEdge;
+    *operandUseEdge = replacement;
+
+    const bool isArgument = (replacement != gtControlExpr) && ((gtCallType != CT_INDIRECT) ||
+        ((replacement != gtCallCookie) && (replacement != gtCallAddr)));
```

Comment:
> What happens when gtCallType == CT_INDIRECT, this seems wrong to me.
Wrong in what way? The `gtCallCookie` and `gtCallAddr` fields aren't relevant unless `gtCallType == CT_INDIRECT` and when they are relevant, AFAIK they are not present in the call argument table.